### PR TITLE
Admin API's and improved support for sub-path mounting

### PR DESCRIPTION
**Pull request by vanpelt**
Description
-----------

A customer asked for api methods to manage teams and users in wandb/local.  This PR adds the following methods to the public API:

```
api.create_team("some_team")
team = api.team("some_team")
team.members[0].delete()
team.invite("username_or_email")
team.create_service_account("My service account")
user = api.user("username_or_email")
user.generate_api_key()
user.delete_api_key("XXXXXX")
```

Testing
-------

Added mock server endpoints and tests

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------
* Invite users to teams via the api `api.team("team_name").invite("username_or_email")`
* Remove users from a team via the api `api.team("team_name").members[0].delete()`
* Create service accounts via the api `api.team("team_name").create_service_account("Description")`
* Manage api keys via the api `api.user("username_or_email").generate_api_key()`
------------- END RELEASE NOTES --------------------

### CODE REVIEW

File: wandb/apis/public.py [hunk 1]
```diff
@@ -662,8 +736,268 @@ def __next__(self):
...
+            True on success
+
+        Raises:
+            requests.exceptions.HTTPError if the user couldn't be found
+        """
+        variables = {"entityName": self.name, "admin": admin}
+        if "@" in username_or_email:
+            variables["email"] = username_or_email
+        else:
+            variables["username"] = username_or_email
```
Comment (kptkin) [inline wandb/apis/public.py:966]: corner case: if username has `@` do we check for this (i.e. is this allowed?)

Comment (vanpelt) [reply wandb/apis/public.py:966]: Usernames can only include letters, dashes, and underscores.

Comment (kptkin) [reply wandb/apis/public.py:966]: got it thanks. 
just for my understanding: If someone uses our public api and provides invalid username - are we only catching it in the backend?

Comment (vanpelt) [reply wandb/apis/public.py:966]: We should be preventing it on user creation and we don't provide an api's to create users currently.

**Pull request closed without merging**